app/controllers/main_routes/roomPreference.py

```diff
@@ -20,55 +20,54 @@ def roomPreference(term):
-                                        .where(Course.offCampusFlag == False)
-                                        .distinct().where(Course.parentCourse == None)
-                            )
-        # roompreferences = RoomPreferences.select().join(Course, on = (RoomPreferences.course == Course.cId)).join(InstructorCourse, on=(Course.cId == InstructorCourse.course)).where(InstructorCourse.username == current_user and Course.term == current_term).distinct()
-
-        return render_template(
-                "roomPreference.html",
-                roompreferences= roompreferences,
-                room=room,
-                users=users,
-                course=courses,
-                educationTech=educationTech,
-                instructors=instructors,
-                cfg = cfg,
-                isAdmin = au.user.isAdmin
-            )
-    else:
-        return render_template("roomPreferencesLocked.html", cfg = cfg,
-                                                             isAdmin = au.user.isAdmin)
+
+    current_user = au.username
+
+    # Used to populate dropdowns and stuff
+    room = Rooms.select().join(Building, on = (Building.bID == Rooms.building)).order_by(Building.name.asc(), Rooms.number.asc())
+
+    users= User.select()
+    instructors = InstructorCourse.select()
+    educationTech= EducationTech.select()
+
+    courses = ( Course.select()
+
+                    .join(InstructorCourse, on= (InstructorCourse.course == Course.cId))
+                    .join(Term, on=(Term.termCode == Course.term))
+                    .where(Course.term == int(current_term))
+                    .where(Course.offCampusFlag == False))
+
+    # Constructs RoomPreferences if they don't exist
+    for course in courses:
+        (rp, c) = RoomPreferences.get_or_create(course = course.cId)
+
+    roompreferences= (  RoomPreferences.select()
+                                    .join(InstructorCourse, on = (InstructorCourse.course == RoomPreferences.course))
+                                    .join(Course, on = (RoomPreferences.course == Course.cId))
+                                    .join(Term, on = (Course.term == Term.termCode))
+                                    .where(RoomPreferences.course == InstructorCourse.course)
+                                    .where(Course.term == current_term)
+                                    .where(Course.offCampusFlag == False)
+                                    .distinct().where(Course.parentCourse == None))
+
+    if au.user.isAdmin == False:
+        roompreferences = roompreferences.where(InstructorCourse.username == current_user)
+        courses = courses.where(InstructorCourse.username == current_user)
+        if term.term_state.number != 3:
+            return render_template("roomPreferencesLocked.html", cfg = cfg,
+                                                                 isAdmin = au.user.isAdmin)
```

Comment:
Whether or not room preferences are locked shouldn't depend on isAdmin. Everyone should see locked when the term isn't in the right state.